WIP using JSON responses to AJAX requests

lainets opened the pull request:
I created this PR to discuss the idea.

I do not think it is a good idea to load pages using ajax and then use DOM manipulation to get the needed information out of the loaded page. This will easily cause issues (that are also hard to notice and debug) if the page HTML changes. There is also no indication in the HTML itself (or whatever creates it) that the JS is dependent on a specific structure. I propose we add JSON versions of the pages that would otherwise need such DOM introspection to move the logic to python where it will be much easier to see if changes have an effect on something else.

# Description

**What?**

[ANSWER HERE]

**Why?**

[ANSWER HERE]

**How?**

[ANSWER HERE]

# Testing

**What type of test did you run?**

- [ ] Accessibility test using the [WAVE](https://wave.webaim.org/extension/) extension.
- [ ] Django unit tests.
- [ ] Selenium tests.
- [ ] Other test. *(Add a description below)*
- [ ] Manual testing.

[ADD A DESCRIPTION ABOUT WHAT YOU TESTED MANUALLY]

**Did you test the changes in**

- [ ] Chrome
- [ ] Firefox
- [ ] This pull request cannot be tested in the browser.

**Think of what is affected by these changes and could become broken**

# Translation

- [ ] Did you modify or add new strings in the user interface? ([Read about how to create translation](https://github.com/apluslms/a-plus/tree/master/doc#running-tests-and-updating-translations))

# Programming style

- [ ] Did you follow our [style guides](https://apluslms.github.io/contribute/styleguides/)?
- [ ] Did you use Python type hinting in all functions that you added or edited? ([type hints](https://docs.python.org/3/library/typing.html) for function parameters and return values)

# Have you updated the README or other relevant documentation?

- [ ] documents inside the doc directory.
- [ ] README.md.
- [ ] Aplus Manual.
- [ ] Other documentation (mention below which documentation).

# Is it Done?

- [ ] Reviewer has finished the code review
- [ ] After the review, the developer has made changes accordingly
- [ ] Customer/Teacher has accepted the implementation of the feature

Mankro commented:
I thought it would be faster to fix the bug with small changes. You are suggesting large refactoring, but we should fix the active element submission feedback polling quickly. If we at least first finish a quick and small solution, then there is more time later on to think about major refactoring.

For example, the `data-poll-url` attribute is already used here:
https://github.com/apluslms/a-plus/blob/564ee0985050288b4b09bcf99deca6b22f50c091/exercise/static/exercise/chapter.js#L605
If `data-ready-url` were added there similarly, would that lead to a quick and simple fix to the urgent issue (#1188)?

lainets commented:
#1192 implements a quick fix to the issue as we need it fixed ASAP. Let's have this PR solely about the ajax refactoring

lainets committed:
Add tools for multiple accepted response mime types

Use `@accepts_mimes(<list of implemented mime types>)` to decorate view
method/function with the accepted types, and then use
`request.expected_mime` to get the type the client expects (the one that
matches best with the implemented mime types).

lainets committed:
Add tools for returning JSON from views

lainets committed:
Use JSON responses with active elements

Also fixes an issue where the submission isn't visible if messages
are present (e.g. "Staff can submit assignments without enrolling.")

lainets commented:
I now made an implementation of JSON responses to AJAX calls for active elements submitting.

The more I think about this, the more I want to try swapping to using React because then most of the rendering happens on the client, simplifying updating of the site considerably.
